### PR TITLE
Add riemann 1x2v PDE

### PR DESCRIPTION
**Pull request by ckendrick**
Please review the [developer documentation](https://github.com/project-asgard/asgard/wiki/developing)
on the wiki of this project that contains help and requirements.

## Proposed changes

Adds the 1x2v version of the Riemann PDE.

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [ ] Bugfix
- [x] New feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- [x] this PR is up to date with current the current state of 'develop'
- [x] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

File: src/pde/pde_riemann_1x2v.hpp [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+  PDE_riemann_1x2v(parser const &cli_input)
+      : PDE<P>(cli_input, num_dims_, num_sources_, num_terms_, dimensions_,
+               terms_, sources_, exact_vector_funcs_, exact_scalar_func_,
+               get_dt_, do_poisson_solve_, has_analytic_soln_, moments_,
+               do_collision_operator_)
+  {
+    param_manager.add_parameter(parameter<P>{"n", n});
+    param_manager.add_parameter(parameter<P>{"u", u});
+    param_manager.add_parameter(parameter<P>{"u2", u2});
+    // param_manager.add_parameter(parameter<P>{"u3", u3});
```
Comment (stefan-schnake) [inline src/pde/pde_riemann_1x2v.hpp]: Delete

File: src/pde/pde_riemann_1x2v.hpp [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+  initial_condition_const(fk::vector<P> const &x, P const t = 0)
+  {
+    ignore(t);
+    fk::vector<P> fx(x.size());
+    std::fill(fx.begin(), fx.end(), P{1.0 / 12.0});
+    return fx;
+  }
+
+  inline static dimension<P> const dim_0 = dimension<P>(
+      -0.6, 0.6, 8, default_degree,
```
Comment (stefan-schnake) [inline src/pde/pde_riemann_1x2v.hpp]: Default to level 7

**Review by stefan-schnake — CHANGES_REQUESTED** [review]

File: src/pde/pde_riemann_1x2v.hpp [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+  // Term 8
+  // d_{v_y}(th q), q = d_{v_y} f
+  // Same as term 7 but order if changed
+
+  inline static std::vector<term<P>> const terms_im_8 = {
+      nu_theta_term, I_diff_im, diff_v_term};
+
+  inline static term_set<P> const terms_ = {terms_ex_1, terms_ex_2, terms_im_1,
+                                            terms_im_2, terms_im_3, terms_im_4,
+                                            terms_im_7, terms_im_8};
```
Comment (stefan-schnake) [inline src/pde/pde_riemann_1x2v.hpp]: I think for clarity these should be made into terms 4,5,6

**Review by stefan-schnake — CHANGES_REQUESTED** [review]

**Review by stefan-schnake — DISMISSED** [review]

**Review by stefan-schnake — DISMISSED** [review]

File: src/pde/pde_riemann_1x2v.hpp [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+  {
+    ignore(x);
+    ignore(time);
+    return -1.0;
+  }
+
+  static P e1_g2(P const x, P const time = 0)
+  {
+    ignore(time);
+    return (x > 0.0) ? x : 0.0;
```
Comment (quantumsteve) [inline src/pde/pde_riemann_1x2v.hpp]:
```suggestion
    return std::max(P{0.}, x);
```

File: src/pde/pde_riemann_1x2v.hpp [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+  {
+    ignore(x);
+    ignore(time);
+    return -1.0;
+  }
+
+  static P e2_g2(P const x, P const time = 0)
+  {
+    ignore(time);
+    return (x < 0.0) ? x : 0.0;
```
Comment (quantumsteve) [inline src/pde/pde_riemann_1x2v.hpp]:
```suggestion
    return std::min(P{0.}, x);
```

File: src/pde/pde_riemann_1x2v.hpp [hunk 6]
```diff
@@ -0,0 +1,499 @@
...
+
+  inline static std::vector<moment<P>> const moments_ = {
+      moment0, moment1v1, moment1v2, moment2v1, moment2v2};
+
+  /* Construct (n, u, theta) */
+  static P n(P const &x, P const t = 0)
+  {
+    ignore(t);
+    P const first  = x < -0.3 ? 1.0 : 0.0;
+    P const second = (x >= -0.3 && x <= 0.3) ? (1.0 / 8.0) : 0.0;
```
Comment (quantumsteve) [inline src/pde/pde_riemann_1x2v.hpp]:
```suggestion
    P const second = std::abs(x) <= 0.3 ? (1.0 / 8.0) : 0.0;
```

File: src/pde/pde_riemann_1x2v.hpp [hunk 7]
```diff
@@ -0,0 +1,499 @@
...
+    ignore(x);
+    return 0.0;
+  }
+
+  static P theta(P const &x, P const t = 0)
+  {
+    ignore(t);
+    ignore(x);
+    P const first  = x < -0.3 ? 1.0 : 0.0;
+    P const second = (x >= -0.3 && x <= 0.3) ? (4.0 / 5.0) : 0.0;
```
Comment (quantumsteve) [inline src/pde/pde_riemann_1x2v.hpp]:
```suggestion
    P const second = std::abs(x) < 0.3 ? (4.0 / 5.0) : 0.0;
```

**Review by quantumsteve — APPROVED** [review]

**Pull request merged**